src/psyclone/psyir/transformations/reference2arrayrange_trans.py

```diff
@@ -161,6 +161,11 @@ def validate(self, node, options=None):
                 lambda: f"References to arrays passed to "
                 f"'{node.parent.routine.name}' intrinsics should not be "
                 f"transformed, but found:\n {node.parent.debug_string()}"))
+        assignment = node.ancestor(Assignment)
+        if assignment and assignment.is_pointer:
+            raise TransformationError(
+                f"'{type(self).__name__}' can not be applied to references"
+                f" inside pointer assignments, but found '{node.name}'")
```

Comment:
"...in {assignment.debug_string()}" perhaps?

Reply:
Done (I think a verbose option would also be useful here but I have another issue to look at refactoring this whole transformation, so I will do it there)